src/lib/selection-impl.ts

```diff
@@ -196,38 +185,20 @@ export class SelectionImpl<T> implements Selection<T> {
         this.boundData[index] = datum;
         this.sprites[index] = sprite;
 
-        const {initCallback, enterCallback, bindCallback} = this;
-
-        if (initCallback || bindCallback) {
-          // Schedule the Sprite's enter() callback to run. This will invoke
-          // the bindCallback and/or the initCallback, in that order.
+        if (initCallback) {
+          // Use Sprite's enter() to invoke initCallback.
           sprite.enter(spriteView => {
-            if (bindCallback) {
-              // The bindCallback, if present is always invoked when binding
-              // data, immediately before more specific callbacks if present.
-              bindCallback(spriteView, datum);
-            }
-            if (initCallback) {
-              initCallback(spriteView, datum);
-            }
+            initCallback(spriteView, datum);
             // NOTE: Because init() applies to the first frame of an entering
             // data point, it should never have a transition time.
             spriteView.TransitionTimeMs = 0;
           });
         }
 
-        if (enterCallback || bindCallback) {
-          // Schedule the Sprite's update() callback to run. This will invoke
-          // the bindCallback and/or the enterCallback, in that order.
+        if (enterCallback) {
+          // Use Sprite's update() to invoke enterCallback.
```

Comment:
Using `update()` to trigger the `enterCallback()` seems odd to me. Can you be more descriptive about why you're doing it that way instead of via `enter()`?

Reply:
Yeah, it's a little confusing. Added inline documentation here, changed some variable names, and added more API docs to `sprite.d.ts`.